src/Microsoft.AspNetCore.StaticFiles/StaticFileMiddleware.cs

```diff
@@ -59,6 +60,7 @@ public StaticFileMiddleware(RequestDelegate next, IHostingEnvironment hostingEnv
             _contentTypeProvider = options.Value.ContentTypeProvider ?? new FileExtensionContentTypeProvider();
             _fileProvider = _options.FileProvider ?? Helpers.ResolveFileProvider(hostingEnv);
             _matchUrl = _options.RequestPath;
+            _responseCacheFilter = new ResponseCacheFilter();
```

Comment:
If there was no intention of making `IResponseCacheFilter` something others will implement why bother with the interface and type at all? I think a method on `StatiFileContext` would be sufficient then.